pkg/controller/podautoscaler/horizontal.go

```diff
@@ -50,8 +51,8 @@ const (
 )
 
 type HorizontalController struct {
-	scaleNamespacer unversionedextensions.ScalesGetter
-	hpaNamespacer   unversionedextensions.HorizontalPodAutoscalersGetter
+	scaleClient   *scaleclient.Client
```

Comment:
Only one you know of. Things using the HPA controller to scale other resources can (and do) provide their own impl. Additionally, we should be testing the controller, which would be far easier with an interface than a required impl